client/driver/docker.go

```diff
@@ -1099,7 +1099,7 @@ func (d *DockerDriver) newDockerClient(timeout time.Duration) (*docker.Client, e
 		}
 	}
 
-	if timeout != 0 {
+	if timeout != 0 && newClient != nil {
```

Comment:
Will the err be set in this case? Otherwise if newClient is nil and the error is nil, it seems like the caller is susceptible to a NPE too

Reply:
Yes the error is always set if newClient is nil. I read through the methods that return a new dockerClient in https://github.com/hashicorp/nomad/blob/5f0f349a114f0e11a6f32636be9cf101be34c9fa/vendor/github.com/fsouza/go-dockerclient/client.go#L39 and it always returns an error if the client is nil.